include/FixedPointConverter.h

```diff
@@ -241,81 +242,88 @@ namespace ChimeraTK {
   };
 
   /********************************************************************************************************************/
-
-  // FIXME: replace reinterpret_cast with bit_cast once C++20 is available for us.
-  // Until then, turn off the linter warning about reinterpret_cast
-  // NOLINTBEGIN(cppcoreguidelines-pro-type-reinterpret-cast)
+  template<typename RawType>
   template<typename UserType, typename RAW_ITERATOR, typename COOKED_ITERATOR>
-  void FixedPointConverter::vectorToCooked_impl<UserType, RAW_ITERATOR, COOKED_ITERATOR>::impl(
+  void FixedPointConverter<RawType>::vectorToCooked_impl<UserType, RAW_ITERATOR, COOKED_ITERATOR>::impl(
       const FixedPointConverter& fpc, const RAW_ITERATOR& raw_begin, const RAW_ITERATOR& raw_end,
       COOKED_ITERATOR cooked_begin) {
     // Handle integer and floating-point types differently.
     switch(boost::fusion::at_key<UserType>(fpc.conversionBranch_toCooked.table)) {
       case 1: { // std::numeric_limits<UserType>::is_integer && _fpc->_fractionalBits == 0 && !_fpc->_isSigned
-        std::transform(raw_begin, raw_end, cooked_begin, [&fpc](int32_t rawValue) {
+        std::transform(raw_begin, raw_end, cooked_begin, [&fpc](RawType rawValue) {
           fpc.padUnusedBits(rawValue);
-          return numericToUserType<UserType>(*(reinterpret_cast<uint32_t*>(&rawValue)));
+          if constexpr(sizeof(RawType) == 4) {
+            return numericToUserType<UserType>(*(std::bit_cast<uint32_t*>(&rawValue)));
+          }
+          else if constexpr(sizeof(RawType) == 8) {
```

Comment:
I am wondering if we should separate the 64 bit case into another commit. 

1. commit: Formally introduce RawType as template parameter, but have an assertion in the constructor that only allows int32_t as type.
2. commit: Test extension to 64 bits, then make it work.

We first get commit 1. ready and put it into the master before we are messing with too may things.